### PR TITLE
Update SAML Endpoint

### PR DESCRIPTION
**Pull request by hpkuppuraj**
I am setting up v2.1 cluster with saml authentication, all the configuration seems to be correct however when i try to login and found out that saml authrequest still builds ACS URL with _opendistro/_security/saml/acs instead of /_plugin/_security/saml/acs.
Below is what we extracted from the saml tracer plugin.

<samlp:AuthnRequest xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
                    xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion"
                    ID="ONELOGIN_XYZ"
                    Version="2.0"
                    IssueInstant="2022-07-13T08:55:10Z"
                    Destination="https://domain/trust/saml2/http-redirect/sso/<uuid>"
                    ProtocolBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST"
                    AssertionConsumerServiceURL="https://domain/_opendistro/_security/saml/acs"
                    >
    <saml:Issuer>opensearch-saml</saml:Issuer>
    <samlp:NameIDPolicy Format="urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified"
                        AllowCreate="true"
                        />
</samlp:AuthnRequest>

Seems like after version 2.1.0 both enpoints: /_opendistro/_security/saml/acs and /_plugins/_security/saml/acs not working.
If using first-one as IDP acs URL getting: {"statusCode":404,"error":"Not Found","message":"Not Found"} - probbably expected after https://github.com/opensearch-project/security-dashboards-plugin/pull/895
But if switching to 2nd one /_plugins/_security/saml/acs getting {"statusCode":500,"error":"Internal Server Error","message":"Internal Error"}

Rolling back Kibana to version 2.0.1 but leaving Opensearch version: 2.1.0 helps the issue a bit as /_opendistro/_security/saml/acs starts to work again

### Description
[Describe what this change achieves]
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
* Bug Fix
* Why these changes are required?
*  Login Dashboard v2.1 with SAML authentication
* What is the old behavior before changes and new behavior after changes?
*  Old behaviour doesnt allow dashboard v2.1 to login with saml authentication 

### Issues Resolved
[List any issues this PR will resolve]

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
@hpkuppuraj Thanks for this pull request, could you please see about fixing the DCO errors, [link](https://github.com/opensearch-project/security/pull/1936/checks?check_run_id=7318769192)?

**Comment by peternied** [comment]
Note; I said that this changed was set against the 2.1 branch, lets merge this against main and then the maintainers will backport the change to the next release version

**Comment by cliu123** [comment]
@hpkuppuraj Thank you so much for the finding and the PR! 
This would be a breaking change with changing things like ACS URL even though it would align the behavior with the documentation. If an old version cluster without this change gets upgraded to a new version with this change, the SAML authN would be broken. So IMHO, this will be a great change for 3.0.0 release.

**Comment by hpkuppuraj** [comment]
@peternied , Thanks for highlighting the signoff part which i have missed at commits. I think i have signed the commits that I have made. Looks like still DCO error there.

**Comment by DarshitChanpura** [comment]
@hpkuppuraj I see there are 93 file changes here and some of them are from other PRs. Can you please rebase your branch against main so that the changes you intended to make are the only ones included.

**Comment by JustinasKO** [comment]
Maybe better to make both endpoints working? Doesn't seem to be much work and then it would not be breaking change. and in version 3.0 remove `_opendistro` one.

**Comment by peternied** [comment]
> Maybe better to make both endpoints working? Doesn't seem to be much work and then it would not be breaking change. and in version 3.0 remove `_opendistro` one.

I would prefer this approach as suggested by @JustinasKO.  @hpkuppuraj is this something you'd be able to pick up or would you like to take this on?  

Note; there is another change like this in flight - https://github.com/opensearch-project/security/pull/1949

**Comment by hpkuppuraj** [comment]
> > Maybe better to make both endpoints working? Doesn't seem to be much work and then it would not be breaking change. and in version 3.0 remove `_opendistro` one.
> 
> I would prefer this approach as suggested by @JustinasKO. @hpkuppuraj is this something you'd be able to pick up or would you like to take this on?
> 
> Note; there is another change like this in flight - #1949

Yes sure, let me explore on this and keep both the endpoints working.

**Comment by cliu123** [comment]
> @hpkuppuraj I see there are 93 file changes here and some of them are from other PRs. Can you please rebase your branch against `main` so that the changes you intended to make are the only ones included.

@hpkuppuraj Would you please follow up with this comment?

**Comment by hpkuppuraj** [comment]
@cliu123 , I am trying to figure how the PR gets the changes from other commits, but my local branch has just my changes. Could you please help on this.

**Comment by JustinasKO** [comment]
> 

@hpkuppuraj just rebase your branch from master branch and only your changes will remain.

**Comment by hpkuppuraj** [comment]
@cliu123   , could you please confirm the changes. I believe now only one file change is shown. Also i am seeing DCO author signature mismatch now

File: src/main/java/com/amazon/dlic/auth/http/saml/Saml2SettingsProvider.java
Comment (nerijus) [inline src/main/java/com/amazon/dlic/auth/http/saml/Saml2SettingsProvider.java:215]: There is no need to have such comment in the source file, it belongs to the commit message.

Comment (DarshitChanpura) [reply src/main/java/com/amazon/dlic/auth/http/saml/Saml2SettingsProvider.java:215]: +1

**Comment by DarshitChanpura** [comment]
> @cliu123 , could you please confirm the changes. I believe now only one file change is shown. Also i am seeing DCO author signature mismatch now

I can confirm that I see only 1 file change now. You can fix the DCO error by executing this command: `git rebase --signoff HEAD~2` where 2 is the number of commits from current HEAD. Looking at your commit history, you've signed-off your latest commit with `Signed-off-by: Hari Prasad Kuppuraj <51482940+hpkuppuraj@users.noreply.github.com>` which is not correct. It should be done with your Github user (which is, I believe, what you did in the commit prior to the latest one).

**Comment by hpkuppuraj** [comment]
@DarshitChanpura , i dont have access to the account(company) i used to commit the first change, hence switched to the personal github account. Hence there is a discrepancy in the signature.

**Comment by peternied** [comment]
@hpkuppuraj Thanks for your hard work on this pull request, I've cleaned up the commit history and fixed the DCO issues and pushed an update onto your fork, as soon as CI passes this can be merged.

**Review by peternied — APPROVED** [review]

**Review by DarshitChanpura — APPROVED** [review]

**Comment by cliu123** [comment]
Should [this PR](https://github.com/opensearch-project/security-dashboards-plugin/pull/1038) be closed if merging this one? How was this change tested?

**Review by cliu123 — CHANGES_REQUESTED** [review]
Thanks for the PR!
I have a few questions here:
1. Should [this PR](https://github.com/opensearch-project/security-dashboards-plugin/pull/1038) be closed if merging this one? 
2. How was this change tested? Please provide information about testing in the PR description.
3. With this change, how would the existing clusters built with the old 
 ACS endpoint(`/_opendistro/_security/saml/acs`) work?

**Comment by DarshitChanpura** [comment]
@hpkuppuraj Can you please update on the requested changes?

**Comment by peternied** [comment]
@cliu123 Can you shepherd this change along with the frontend to make sure everything is in alignment?

**Comment by hpkuppuraj** [comment]
Hi, apologies I am outstation and will be back to workstation tomorrow. Will resolve the code conflicts

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by peternied** [comment]
@opensearch-project/security this is a breaking change we've been considering for a while how do you feel about us taking it?  I'm leaning towards no, while it would be nice to have consistent URLs, I would prefer we don't break folks in a hard to detect way.

Please vote 👍 👎  or comment on this issue, lets revisit this during our next triage.

**Comment by peternied** [comment]
I'm closing this pull request out as we aren't taking steps to get this merged.  Thanks again for the contribution sorry it didn't work out on this change.

**Comment by nerijus** [comment]
Pity. Then `opensearch-dashboards` should be patched, because now we have to use every time we update it:
```
sed -i 's/_plugins/_opendistro/g' /usr/share/opensearch-dashboards/plugins/securityDashboards/server/auth/types/saml/routes.js
```

**Comment by peternied** [comment]
@nerijus from your comment it sounds SAML is broken in security-dashboards-plugin without this change? Can you create an issue

**Comment by nerijus** [comment]
Of course it is broken, that's why this PR was created. I'd still suggest applying this PR instead of changing dashboards.

**Comment by peternied** [comment]
@cliu123 Could you shepherd this PR and determine how to follow up? This seems very bad (tm) and I'm out of the loop on the state of the SAML configuration in dashboards.

**Comment by aoguan1990** [comment]
@cliu123 As per discussion, Endpoint: `/_plugin/_security/saml/acs` is also used by Dashboards Security Plugin. Endpoint changed should be planed for both front-end and back-end plugin together.

**Comment by cliu123** [comment]
This is a documentation issue that I'm creating. Users follow the documentation to configure `_plugins/_security/saml/acs` in IDP, but in plugins, it still uses `/_opendistro/_security/saml/acs`.

**Comment by cliu123** [comment]
This is a documentation issue that I'm creating. Users follow the documentation to configure _plugins/_security/saml/acs in IDP, but in plugins, it still uses /_opendistro/_security/saml/acs.

**Comment by cwperks** [comment]
@cliu123 The documentation website was reverted to reflect this: https://github.com/opensearch-project/documentation-website/pull/877

There was a breaking change to SAML introduced in 2.2 that was reverted in 2.3 with this PR: https://github.com/opensearch-project/security-dashboards-plugin/pull/895

The frontend switched to using `/_plugins` while this backend change was not applied.

We reverted to using `/_opendistro/_security/saml/acs` for the 2.x line, but it was my understanding that we are still planning this change for 3.0.0. Now that our main branch is building 3.0.0, what is preventing us from merging this change?

**Pull request closed without merging**
